include/zephyr/net/net_ip.h

```diff
@@ -853,6 +866,18 @@ static inline bool net_ipv6_is_ula_addr(const struct in6_addr *addr)
 	return addr->s6_addr[0] == 0xFD;
 }
 
+/**
+ * @brief Check if the given IPv6 address is a global address.
+ *
+ * @param addr A valid pointer on an IPv6 address
+ *
+ * @return True if it is, false otherwise.
+ */
+static inline bool net_ipv6_is_global_addr(const struct in6_addr *addr)
+{
+	return addr->s6_addr[0] == 0x20;
```

Comment:
Shoudl `addr->s6_addr[0]` be ored with `0xE0` before comparing? Only the first 3 bits indicate address scope, current logic will incorrectly filter out global addresses other than thise starting with `0x20`.

Reply:
I think we have to use and and not or.

Reply:
Of course, you're right, I don't know what I was thinking. We need to &.

Reply:
Done